### PR TITLE
BeamAgent

### PR DESCRIPTION
**Pull request by dlinman**

### CODE REVIEW

File: GameEngineTests/Agents/BeamAgentTests.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
+﻿using System.Linq;
+using GameEngine;
+using GameEngine.Agents;
+using GameEngine.Heuristics;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace GameEngineTests.Agents
+{
+    [TestClass]
+    public class BeamAgentTests
```
Comment (rpherbig) [inline GameEngineTests/Agents/BeamAgentTests.cs:10]: Why is there no unit test for the agent formulating a play?

**Review by rpherbig — APPROVED** [review]

File: GameEngineTests/Agents/BeamAgentTests.cs (hunk 1 above)
Comment (dlinman) [reply GameEngineTests/Agents/BeamAgentTests.cs:10]: We can certainly check to see if a legitimate play object comes back from the FormulatePlay method, however I don't think there's a good way to verify that any of the beam searches properties are functioning as intended.

Do you want just like a simple check to make sure that *some* play comes back or are you asking for something more complicated?

Comment (rpherbig) [reply GameEngineTests/Agents/BeamAgentTests.cs:10]: I'm fine with a simple check

Comment (dlinman) [reply GameEngineTests/Agents/BeamAgentTests.cs:10]: 👍

**Comment by rpherbig** [comment]
@DrSeabass - want to review this?

**Pull request merged**